propagation.go

```diff
@@ -41,71 +46,124 @@ var (
 type BuiltinFormat byte
 
 const (
-	// SplitBinary encodes the Span in a SplitBinaryCarrier instance.
+	// Binary encodes the Span for propagation as opaque binary data.
+	//
+	// For Tracer.Inject(): the carrier must be an `io.Writer`.
 	//
-	// The `carrier` for injection and extraction must be a
-	// `*SplitBinaryCarrier` instance.
-	SplitBinary BuiltinFormat = iota
+	// For Tracer.Join(): the carrier must be an `io.Reader`.
+	Binary BuiltinFormat = iota
 
-	// SplitText encodes the Span in a SplitTextCarrier instance.
+	// TextMap encodes the Span in a TextMapCarrier instance.
+	//
+	// For Tracer.Inject(): the carrier must be a `TextMapWriter`.
+	//
+	// For Tracer.Join(): the carrier must be a `TextMapReader`.
+	//
+	// See HTTPHeaderTextMapCarrier for an implementation of both TextMapWriter
+	// and TextMapReader that defers to an http.Header instance for storage.
+	// For example, Inject():
+	//
+	//    carrier := HTTPHeaderTextMapCarrier{
```

Comment:
I got rid of the `util.go` stuff because (a) it really didn't belong in the base library, and (b) after this change it was nothing more than these five lines.